public_html/noaa_cron.php

```diff
@@ -18,75 +18,112 @@
+        
+        // Calculate days out
+        $days_out = (strtotime($date) - strtotime($today)) / (60 * 60 * 24);
+        
+        /////////////////////////DEBUG/////////////////////////
+        // Add each day's data to our debug collection
+        $debug_all_data[$code][] = [
+            'date' => $date,
+            'high' => $high,
+            'low' => $low,
+            'conditions' => $conditions,
+            'icon' => $icon,
+            'days_out' => $days_out
+        ];
+        /////////////////////////DEBUG/////////////////////////
+        
+        $query_insert = "INSERT INTO noaa_weather 
                     (location_code, time_retrieved, forecast_create_date, forecast_for_date,
                      forecast_days_out, forecast_high, forecast_low, fc_text, fc_icon_url)
-                 VALUES ( '%s', '%s', '%s', '%s', DATEDIFF( '%s', '%s' ), %s, %s, '%s', '%s')",
-                    mysqli_real_escape_string($link, $code), $now,
-                    $today, $date, $date, $today, $fc_high, $fc_low,
-                    mysqli_real_escape_string($link, $fc_text),
-                    $fc_icon);
-
-            if(!mysqli_query($link, $sql)) {
-                error_log("Location: $code ($date)");
-                error_log('Error: ' . mysqli_error($link));
-                // error_log($sql);
-            } else {
-                // error_log("Location: $code ($date)");
-                // error_log("$fc_high - $fc_low - $fc_text - $fc_icon");
+                         VALUES 
+                         ('$code', '$now', '$today', '$date', $days_out, $high, $low, '$conditions', '$icon')";
+
+        $insert_result = mysqli_query($link, $query_insert);
+        if (!$insert_result) {
+            error_log("Error inserting data for $code on $date: " . mysqli_error($link));
             }
         }
 
-        // location_code, forecast_create_date, forecast_for_date, forecast_days_out, forecast_high, forecast_low, fc_text, fc_text_fog, fc_text_haze, fc_text_hot, fc_text_cold, fc_text_wind, fc_text_rain_chance, fc_text_snow_chance, fc_text_tstorm_chance, fc_text_sky_condition, fc_icon_url, fc_icon_fog, fc_icon_haze, fc_icon_hot, fc_icon_cold, fc_icon_wind, fc_icon_rain_chance, fc_icon_snow_chance, fc_icon_tstorm_chance, fc_icon_sky_condition, actual_high, actual_low, actual_precip, validity_code
-
-            // error_log("Location: $code ($today)");
-            //
-            // error_log("Highs: " . var_export($highs, 1));
-            // error_log("Lows: " . var_export($lows, 1));
-            // error_log("Icons: " . var_export($icons, 1));
-            // error_log("Text: " . var_export($text, 1));
-
-    }
+    $locations_processed++;
 }
 
-echo $now . ": ran OK";
-echo "<br>";
+/////////////////////////DEBUG/////////////////////////
+// Save all debug data for reference
+file_put_contents('/tmp/noaa_cron_processed_data.json', json_encode($debug_all_data, JSON_PRETTY_PRINT));
```

Comment:
It might be useful to log this output rather than writing it to a file (or in addition to writing to a file). In production, it would be useful to capture the processed data in the logs for troubleshooting purposes.